autogpt/agent.py

```diff
@@ -26,163 +26,178 @@ class Agent:
-                            )
-                            continue
-                        break
-                    elif console_input.lower() == "n":
-                        self.user_input = "EXIT"
-                        break
-                    else:
-                        self.user_input = console_input
-                        command_name = "human_feedback"
-                        break
-
-                if self.user_input == "GENERATE NEXT COMMAND JSON":
-                    logger.typewriter_log(
-                        "-=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-=",
-                        Fore.MAGENTA,
-                        "",
-                    )
-                elif self.user_input == "EXIT":
-                    print("Exiting...", flush=True)
-                    break
-            else:
-                # Print command
-                logger.typewriter_log(
-                    "NEXT ACTION: ",
-                    Fore.CYAN,
-                    f"COMMAND = {Fore.CYAN}{command_name}{Style.RESET_ALL}"
-                    f"  ARGUMENTS = {Fore.CYAN}{arguments}{Style.RESET_ALL}",
-                )
+    
+        self.loop_count = 0
+        while not self.exit_loop:
 
-            # Execute command
-            if command_name is not None and command_name.lower().startswith("error"):
-                result = (
-                    f"Command {command_name} threw the following error: {arguments}"
-                )
-            elif command_name == "human_feedback":
-                result = f"Human feedback: {self.user_input}"
-            else:
-                result = (
-                    f"Command {command_name} returned: "
-                    f"{cmd.execute_command(command_name, arguments)}"
-                )
-                if self.next_action_count > 0:
-                    self.next_action_count -= 1
+            self.loop_count += 1
 
-            memory_to_add = (
-                f"Assistant Reply: {assistant_reply} "
-                f"\nResult: {result} "
-                f"\nHuman Feedback: {self.user_input} "
-            )
+            response = self.start_argent()
 
-            self.memory.add(memory_to_add)
+            if response == 'continue':
+                continue
 
-            # Check if there's a result from the command append it to the message
-            # history
-            if result is not None:
-                self.full_message_history.append(create_chat_message("system", result))
-                logger.typewriter_log("SYSTEM: ", Fore.YELLOW, result)
-            else:
-                self.full_message_history.append(
-                    create_chat_message("system", "Unable to execute command")
-                )
-                logger.typewriter_log(
-                    "SYSTEM: ", Fore.YELLOW, "Unable to execute command"
-                )
+            self.check_limit_loop_count()
+
+    def start_argent(self):
```

Comment:
Should this be start_agent?

```suggestion
    def start_agent(self):
```